applications/ParticleMechanicsApplication/custom_utilities/mpm_particle_generator_utility.h

```diff
@@ -50,11 +50,253 @@ namespace MPMParticleGeneratorUtility
      * @brief Construct material points or particles from given initial mesh
      * @details Generating particles using a designated shape functions
      */
+    template<SizeType TDimension>
     void GenerateMaterialPointElement(  ModelPart& rBackgroundGridModelPart,
                                         ModelPart& rInitialModelPart,
                                         ModelPart& rMPMModelPart,
                                         bool IsAxisSymmetry = false,
-                                        bool IsMixedFormulation = false);
+                                        bool IsMixedFormulation = false) {
+        // Initialize zero the variables needed
+        std::vector<array_1d<double, 3>> xg = { ZeroVector(3) };
```

Comment:
perhaps we should rename xg to mp_xg so that it is similar to the other variables ?

Reply:
I fully agree that we should discuss this. Should we maybe go for calling it material_point_coordinates - something like that?

Reply:
I would also use the abbreviation "mp" - perhaps "mp_coordinates" or "mp_global_coordinates" ?